ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/methods/EeaJsonRpcMethods.java

```diff
@@ -42,13 +43,17 @@ public EeaJsonRpcMethods(
   protected Map<String, JsonRpcMethod> create(
       final PrivacyController privacyController,
       final EnclavePublicKeyProvider enclavePublicKeyProvider) {
-    return mapOf(
-        new EeaSendRawTransaction(
-            getTransactionPool(),
-            privacyController,
-            enclavePublicKeyProvider,
-            getPrivacyParameters().isOnchainPrivacyGroupsEnabled()),
-        new PrivGetEeaTransactionCount(privacyController, enclavePublicKeyProvider));
+    if (getPrivacyParameters().isOnchainPrivacyGroupsEnabled()) {
```

Comment:
It would be good to have a unit test checking that we instantiate the correct rpc method based on the onchain flag.

Reply:
This is tested in the privacy ATs. To do that in this class would be quite a bit of code ...